pageserver/src/tenant.rs

```diff
@@ -1181,46 +1255,35 @@ impl Tenant {
                             // Also, maybe we'll remove that option entirely in the future, see https://github.com/neondatabase/neon/issues/4099.
                             info!("is_deleted is set on remote, resuming removal of timeline data originally done by timeline deletion handler");
 
+                            if !found_delete_mark {
+                                // Can happen because we create delete mark after we set deleted_at in the index part.
+                                self.create_timeline_delete_mark(timeline_id)
+                                    .context("create delete mark")
+                                    .map_err(LoadLocalTimelineError::ResumeDeletion)?;
+                            };
```

Comment:
`create_timeline_delete_mark` can be idempotent. Use `create_new` and detect "already exists" case to avoid the fsync. Although, technically, you never know whether a file that exists has actually been fsynced in the past. So maybe just fsync anyways.

Also, why can't an idempotent `create_timeline_delete_mark` be part of `resume_deletion`?

Reply:
Not sure I fully understand how create_new allows to avoid fsync. Anyway if you created a file you need to fsync it and its parent directory.

> Also, why can't an idempotent create_timeline_delete_mark be part of resume_deletion?

It can, though here we already know that it exists and already make decisions based on that, so I decided to just avoid creating it when there is no need to do so.

If you really insist I can change it. Doesnt make a big difference to me. I can add it to the scope of possible follow up I mentioned here https://github.com/neondatabase/neon/pull/4552#discussion_r1253520946

Reply:
> Not sure I fully understand how create_new allows to avoid fsync. Anyway if you created a file you need to fsync it and its parent directory.

It doesn't. My train of thought was that you didn't make it idempotenat because you wanted to avoid the perf overhead of creating the marker file and fsyncing.

I think less `if` is always better, less conditions to keep track off.

Reply:
Note that below we do completely different actions based on `found_delete_mark` (for local-only case). I removed the if there and folded it into `resume_deletion`. When doing that I realized that there was duplicated method on tenant: `self.create_timeline_delete_mark` so removed it.

(resolving to clearly see remaining items, feel free to reopen)